### PR TITLE
#504: Add Support for Spotinst Elastigroup Component

### PR DESCRIPTION
**Pull request by danieldop**
This PR fixes: https://github.com/zalando-stups/senza/issues/506

### CODE REVIEW

File: examples/elastigroup.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        scaling: {}
+        compute:
+          instanceTypes:
+            ondemand: m3.large
+            spot:
+            - m3.large
+            - m4.large
+            - c3.large
+            - c4.large
+          availabilityZones:
```
Comment (lmineiro) [inline examples/elastigroup.yml]: Could this also work with the [auto-configuration](http://stups.readthedocs.io/en/latest/components/senza.html#senza-stupsautoconfiguration) (discovery of subnets)

File: examples/elastigroup.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          - name: us-west-2a
+            subnetId: subnet-79daaaaa
+          - name: us-west-2b
+            subnetId: subnet-1ba25aaa
+          launchSpecification:
+            monitoring: false
+            imageId: ami-32d81aaa
```
Comment (lmineiro) [inline examples/elastigroup.yml]: Could this also work with the auto-configuration (discovery of latest AMI)

File: examples/elastigroup.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            keyPair: test-key
+            securityGroupIds:
+            - sg-5750aaaa
```
Comment (lmineiro) [inline examples/elastigroup.yml]: Will it accept the security group name?

File: examples/elastigroup.yml (hunk 2 above)
Comment (danieldop) [reply examples/elastigroup.yml]: I'm looking into the auto-configuration.

File: examples/elastigroup.yml (hunk 3 above)
Comment (danieldop) [reply examples/elastigroup.yml]: no, do you guys make them interchangeable?

Comment (lmineiro) [reply examples/elastigroup.yml]: We only specify the security group names

File: examples/elastigroup.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+SenzaInfo:
+  StackName: hello-world
+  StackVersion: one
+
+SenzaComponents:
+  - AppServerElastigroup:
+      Type: Spotinst::Elastigroup
+      "ServiceToken": "arn:aws:lambda:us-west-2:178579023202:function:spotinst-cloudformation"
```
Comment (lmineiro) [inline examples/elastigroup.yml]: What exactly is this value? Is it a well known static value needed for this to work or just an example?

**Comment by lmineiro** [comment]
Just to be sure, this would also work when using the SpotInst on-premise appliance, right?

**Comment by femueller** [comment]
@danieldop Not a Senza expert myself, but from what I can see it looks good to me.
I'd first recommend to add the docstrings for now.

@jmcs @tuxlife Can you add your 2 cents, please?

**Comment by jmcs** [comment]
Hi, please fill out the pull request description and had some documentation to your functions.

**Comment by jmcs** [comment]
Please also fill out the PR description.

**Comment by jmcs** [comment]
:+1:

**Comment by danieldop** [comment]
Anything missing?

**Comment by femueller** [comment]
@danieldop I did the missing changes and it looks good to me too now
@jmcs Can we merge it now?

**Comment by femueller** [comment]
👍

**Pull request merged**